### ISSUE TITLE
Unpredictable cider-switch-to-repl-buffer behavior

### ISSUE THREAD

**Issue by achikin** [clojure-emacs/cider#2246]
## Expected behavior
When I have several windows side-by-side and use `cider-switch-to-repl-buffer` I expect repl buffer to open in the same window where I issued the command.
## Actual behavior
Buffer opens in random places. See gifs. I understand that there is some logic behind it, but it's hard to predict in which window does the repl buffer go.
![cider-1](https://user-images.githubusercontent.com/611419/37727796-c63e7d7c-2d49-11e8-8e49-657491a39cf0.gif)
![cider-2](https://user-images.githubusercontent.com/611419/37727806-c9f9408c-2d49-11e8-908d-9428af151bfa.gif)

## Steps to reproduce the problem
- `M-x cider-jack-in-clojurescript` and wait for repls to start
- Split screen vertically
- Switch between windows and call `cider-switch-to-repl-buffer` in different windows
- Mention in which window does the repl appear (always the same)
- Open repl buffer in the different window
- Again switch between windows and call `cider-switch-to-repl-buffer` in different windows
- Mention that now it opens always in the other window than current.
*This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.*

### CIDER version information

*Include here the version string displayed when
CIDER's REPL is launched. Here's an example:*

```
CIDER 0.17.0snapshot (package: 20180319.1204), nREPL 0.2.13
;; Clojure 1.8.0, Java 1.8.0_20
```

### Lein/Boot version

Leiningen 2.8.1 on Java 1.8.0_20 Java HotSpot(TM) 64-Bit Server VM

### Emacs version

GNU Emacs 25.3.1

### Operating system

High Sierra 10.13.3(17D102)

**Comment by bbatsov** [comment]
Makes sense to me. Basically want the REPL buffer to open appear in the current window, right? The default behaviour is prompted mostly by the fact that usually you want to be able to see the code on one side and the repl next to it. We leave it to Emacs to decide which window to utilize exactly for the repl.

**Comment by achikin** [comment]
@bbatsov "code on one side and the repl next to it" make sense to me, but in some layouts "next to it" is hard to determine, so for the sake of consistency I'd prefer "in the same window".

**Comment by achikin** [comment]
But yes, it'd be more convinient if the repl opens always *not* in the same window

**Comment by vspinu** [comment]
Last time I checked this was due to an oversight in display-buffer actions in core emacs. Particularly `display-buffer-in-previous-window`, which ends up being called, doesn't handle `(inhibit-same-window . t)`. 

So, basically if you had your REPL being displayed last time in the same window as your current buffer, then your current buffer will be overwritten by the REPL (super annoying, agreed).  

As to a more general concern of having repl in one predefined position, there are specialized packages for that. For instance [window-purpose](http://melpa.org/#/window-purpose) and [frame-purpose](https://github.com/alphapapa/frame-purpose.el) (which is being pushed right now to melpa).
